### PR TITLE
Core Infrastructure Initiative Best Practices status

### PR DESCRIPTION
**Pull request by lgritz**
Copied from here: https://github.com/coreinfrastructure/best-practices-badge/blob/master/doc/criteria.md

I added the colorful emoji annotations next to each item. Most are
marked with the 'unknown' question mark indicating that somebody more
knowledgeable needs to assess it, but I tried to mark the ones as done
(green check box) that were obvious to me.

The idea is that once we check this in to the developer docs, people can
submit PRs that modify the status of items that they know about, or as
they are fixed.

[ci skip]

### CODE REVIEW

File: docs/developers/CII-BestPractices-status.md [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+<b><i>New functionality testing</i></b>
+
+<ul>
+
+<li><a name="test_policy"></a>The project MUST have a general policy (formal or not) that as major new functionality is added to the software produced by the project, tests of that functionality should be added to an automated test suite. <sup>[<a href="#test_policy">test_policy</a>]</sup><dl><dt><i>Details</i>:<dt> <dd>As long as a policy is in place, even by word of mouth, that says developers should add tests to the automated test suite for major new functionality, select "Met."</dd></dl></li>
+
+❓**OCIO needs somebody to assess**
+
+<li><a name="tests_are_added"></a>The project MUST have evidence that the <a href="#test_policy">test_policy</a> for adding tests has been adhered to in the most recent major changes to the software produced by the project. <sup>[<a href="#tests_are_added">tests_are_added</a>]</sup><dl><dt><i>Details</i>:<dt> <dd>Major functionality would typically be mentioned in the release notes. Perfection is not required, merely evidence that tests are typically being added in practice to the automated test suite when new major functionality is added to the software produced by the project.</dd></dl></li>
+
```
Comment (hodoulp) [inline docs/developers/CII-BestPractices-status.md]: I would say that there is an implicit rule (i.e. not documented) which implies to have unit tests for any new feature.

File: docs/developers/CII-BestPractices-status.md [hunk 2]
```diff
@@ -0,0 +1,742 @@
...
+❓ **OCIO needs to assess**
+
+<li><a name="know_common_errors"></a>At least one of the project's primary developers MUST know of common kinds of errors that lead to vulnerabilities in this kind of software, as well as at least one method to counter or mitigate each of them. <sup>[<a href="#know_common_errors">know_common_errors</a>]</sup><dl><dt><i>Details</i>:<dt> <dd>Examples (depending on the type of software) include SQL injection, OS injection, classic buffer overflow, cross-site scripting, missing authentication, and missing authorization. See the <a href="http://cwe.mitre.org/top25/">CWE/SANS top 25</a> or <a href="https://www.owasp.org/index.php/Category:OWASP_Top_Ten_Project">OWASP Top 10</a> for commonly used lists.</dd></dl></li>
+
+❓ **OCIO needs to assess**
+
+</ul>
+
+
+<b><i>Use basic good cryptographic practices</i></b>
```
Comment (hodoulp) [inline docs/developers/CII-BestPractices-status.md]: Is that pertinent for OCIO?

**Comment by lgritz** [comment]
What I'd like to do with this PR is have everybody agree merely that committing this document is the right thing to do to keep track of our progress towards compliance.

We shouldn't haggle over the line-items on this PR... subsequent PRs can edit the document to fill in the details more correctly, or to fix what we need to fix.

File: docs/developers/CII-BestPractices-status.md [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+
+✅ **OCIO done**
+
+<li><a name="test_most"></a>It is SUGGESTED that the test suite cover most (or ideally all) the code branches, input fields, and functionality. <sup>[<a href="#test_most">test_most</a>]</sup></li>
+
+❓**OCIO needs somebody to assess**
+
+<li><a name="test_continuous_integration"></a>It is SUGGESTED that the project implement continuous integration (where new or changed code is frequently integrated into a central code repository and automated tests are run on the result). <sup>[<a href="#test_continuous_integration">test_continuous_integration</a>]</sup><dl><dt><i>Rationale</i>:<dt> <dd>See <a href="http://martinfowler.com/articles/continuousIntegration.html">Martin Fowler</a> There has been some shift in the meaning of the term continuous integration. Historically the term continuous integration focused on the first part - the frequent integration - and not on its testing.  However, over time the emphasis has shifted to include the notion of running automated tests as soon as the code is integrated.  We realize that this can be difficult for some projects to apply, which is why it is only SUGGESTED at the passing level.
+</dd></dl></li>
+
```
[the remark this answers is not in the document]
Comment (hodoulp) [reply docs/developers/CII-BestPractices-status.md]: Agree but that's now more and more mandatory for any project (as well as code coverage, and static analysis) to assess the project’s 'quality'. May we hope to have support with ASWF CI infrastructure?

**Pull request merged**